kedro_sagemaker/cli.py

```diff
@@ -341,5 +347,34 @@ def execute(ctx: CliContext, pipeline: str, node: str, params: str):
     with KedroContextManager(
         ctx.metadata.package_name, env=ctx.env, extra_params=parameters
     ) as mgr:
+        if is_mlflow_enabled():
+            env_key, env_value = lookup_mlflow_run_id(
+                mgr.context, os.getenv(KEDRO_SAGEMAKER_EXECUTION_ARN)
+            )
+            click.echo(f"Mlflow run id: {env_value}")
+            os.environ[env_key] = env_value
+
         runner = SageMakerPipelinesRunner()
         mgr.session.run(pipeline, node_names=[node], runner=runner)
+
+
+@sagemaker_group.command(hidden=True)
+@click.pass_obj
+def mlflow_start(ctx: CliContext):
+    """
+    Registers new mlflow run with Sagemaker Execution ARN inside the tags
+    """
+    import mlflow
+    from kedro_mlflow.config.kedro_mlflow_config import KedroMlflowConfig
+
+    with KedroContextManager(ctx.metadata.package_name, env=ctx.env) as mgr:
+        mlflow_conf: KedroMlflowConfig = mgr.context.mlflow
+
+    run = mlflow.start_run(
```

Comment:
Shouldn't this be nested under `with`?

Reply:
No need, we need a context only to parse the proper kedro-mlflow's config files. Rest of operations can happen outside Kedro